scripts/migrateHistoricBids.ts

```diff
@@ -0,0 +1,118 @@
+import * as fs from "fs";
+import * as env from "env-var";
+import * as dotenv from "dotenv";
+import * as mongodb from "mongodb";
+import { Bid } from "../src/types";
+import { MongoClientOptions } from "mongodb";
+import { queueAdapters, MessageQueueService } from "../src/messagequeue";
+import { CreateBatchOptions } from "@azure/event-hubs";
+
+import {
+  MessageType,
+  TypedMessage,
+  BidPlacedV1Data,
+  BidCancelledV1Data,
+} from "@zero-tech/zns-message-schemas";
+dotenv.config();
+
+const outputFilename = "./output/zAuctionBidsHistorical.json";
+const user = env.get("MONGO_USERNAME").required().asString();
+const pass = env.get("MONGO_PASSWORD").required().asString();
+const uri = env.get("MONGO_CLUSTER_URI").required().asString();
+const dbName = env.get("MONGO_DB").required().asString();
+const collectionName = env.get("MONGO_COLLECTION").required().asString();
+const fullUri = `mongodb+srv://${user}:${pass}@${uri}/`;
+const fileOption = process.argv[2] ?? false;
+const options: MongoClientOptions = {
+  connectTimeoutMS: 5000,
+  w: "majority",
+};
+const main = async () => {
+  const client = new mongodb.MongoClient(fullUri, options);
+  await client.connect();
+  const database = client.db(dbName);
+  const bidsCol = database.collection(collectionName);
+  const cancelCol = database.collection(collectionName + "-archive");
+
+  const bidsPlaced = await bidsCol.find<Bid>({}).toArray();
+  console.log(`${bidsPlaced.length} Bids placed retrieved from DB`);
+  const bidsCancelled = await cancelCol.find<Bid>({}).toArray();
+  console.log(`${bidsCancelled.length} Bids cancelled retrieved from DB`);
+  client.close();
```

Comment:
Consider breaking this logic into its own function(s). I see a reusable function here with a signature that looks something like:

```Typescript
 function getAllFromCollection<T>(collectionName, databaseClient): T[]
```

Reply:
Split functionality into reusable methods